supplier_product/models.py

```diff
@@ -0,0 +1,67 @@
+from django.db import models
+from django.core.validators import MinLengthValidator, MinValueValidator
+from product.models import Product
+
+
+class User(models.Model):
+    user_name = models.CharField(max_length=32,
+                                 primary_key=True,
+                                 null=False,
+                                 validators=[MinLengthValidator(6)],
+                                 help_text="User Name")
+    first_name = models.CharField(max_length=32,
+                                  validators=[MinLengthValidator(1)],
+                                  help_text="First Name")
+    last_name = models.CharField(max_length=32,
+                                 validators=[MinLengthValidator(1)],
+                                 help_text="Last Name")
+    password = models.CharField(max_length=16,
+                                validators=[MinLengthValidator(6)],
+                                null=False,
+                                help_text="Password")
+    user_type = models.CharField(max_length=16,
+                                 validators=[MinLengthValidator(6)],
+                                 null=False,
+                                 choices=[('client', 'client'), ('supplier', 'supplier')],
+                                 help_text="User Type - supplier/client")
+
+    class Meta:
+        abstract = True
+
+
+class Supplier(User):
+    """Supplier class inherit from User abstract class.
+    Supplier class lets the app user to do variety of actions
+    which evolves the Product and Delivery class.
+    Attributes:
+        business_name: a string which holds the Supplier business name.
+    """
+    business_name = models.CharField(max_length=32,
+                                     validators=[MinLengthValidator(6)],
+                                     null=False,
+                                     help_text="Business Name")
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey('Supplier', on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField(validators=[MinValueValidator(1)])
+
+    class Meta:
+        unique_together = ('suppler_product_id', 'qr_code')
+
+    def increase_sup_product_quantity(self):
+        self.quantity += 1
+        self.save()
+
+    def decrease_sup_product_quantity(self):
+        if self.quantity == 0:
+            exit
+        else:
+            self.quantity -= 1
+            self.save()
+
+    def delete_sup_product(self):
+        SupplierProduct.delete()
```

Comment:
You can't call delete() on a class name. Also, you are not calling this function in your tests (you are directly calling `delete()`).
So there is no need for this function.

Reply:
rawan what you can do to make delete_sup_product relevant is to check whether the instance exists, if there even point to use the built in method delete, otherwise like nofar said, not point of warping the built in delete method.

Reply:
@RawanAbuleil When you modify something that has a comment related to it,reply to the comment and resolve if the thing is fixed

Reply:
@RawanAbuleil deleted this method as I suggested, then I'm resolving this comment.
I would also accept Paz's suggestion, so if you want to add that I'd be OK with that.